gpii/node_modules/flowManager/src/UserLogonStateChange.js

```diff
@@ -164,9 +179,21 @@ fluid.defaults("gpii.flowManager.userLogonHandling.stateChangeHandler", {
 fluid.defaults("gpii.flowManager.userLogonHandling.matchMakingStateChangeHandler", {
     gradeNames: ["gpii.flowManager.userLogonHandling.stateChangeHandler",
                  "gpii.flowManager.nonPrivacyFilteredMatchMakingRequest"],
+    events: {
+        onContextAdded: null,
+        onLifecycleInstructions: null
+    },
     listeners: {
         "onUserToken.getDeviceContext": "{that}.getDeviceContext",
-        "onMatchDone.startLifecycle": "{that}.startLifecycle"
+        "onMatchDone.runContextManager": {
+            listener: "{flowManager}.contextManager.addActiveContext",
+            args: ["{arguments}.0", "{that}.events.onContextAdded"]
+        },
+        "onContextAdded.addLifecycleInstructions": {
```

Comment:
Messy duplication of stuff in UntrustedFlowManager - this can be tidier by i) the comment earlier about just operating a single event, ii) factoring this stuff into a new base grade, iii) both